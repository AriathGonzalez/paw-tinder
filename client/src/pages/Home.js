import Nav from "../components/Nav";
import AuthModal from "../components/AuthModal";
import { useState } from "react";
import { useCookies } from "react-cookie";
import "../styles/Homepage.css";

const Home = () => {
  const [showModal, setShowModal] = useState(false);
  const [isSignUp, setIsSignUp] = useState(true);
  const [cookies, , removeCookie] = useCookies(["user"]);
  const authToken = cookies.AuthToken;

  const handleClick = () => {
    if (authToken) {
      removeCookie("UserId", cookies.UserId);
      removeCookie("AuthToken", cookies.AuthToken);
      window.location.reload();
      return;
    }
    setShowModal(true);
    setIsSignUp(true);
  };

  return (
    <div className="overlay">
      <Nav
        authToken={authToken}
        minimal={false}
        showModal={showModal}
        setShowModal={setShowModal}
        setIsSignUp={setIsSignUp}
      />
      <div className="home">
        <h1 className="primary-title">Sniff RightÂ®</h1>
        <button className="primary-button" onClick={handleClick}>
          {authToken ? "Signup" : "Create Account"}
        </button>
        {showModal && (
          <AuthModal
            setShowModal={setShowModal}
            isSignUp={isSignUp}
            setIsSignUp={setIsSignUp}
          />
        )}
      </div>
    </div>
  );
};

export default Home;
