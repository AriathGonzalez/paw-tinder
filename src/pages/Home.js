import Nav from "../components/Nav";
import AuthModal from "../components/AuthModal";
import { useState } from "react";
import "../styles/Homepage.css";

const Home = () => {
  const [showModal, setShowModal] = useState(false);
  const [isSignUp, setIsSignUp] = useState(true);

  const authToken = false;

  const handleClick = () => {
    console.log("home, create account clicked!");
    setShowModal(true);
  };

  return (
    <div className="overlay">
      <Nav
        minimal={false}
        showModal={showModal}
        setShowModal={setShowModal}
        setIsSignUp={setIsSignUp}
      />
      <div className="home">
        <h1 className="primary-title">Sniff RightÂ®</h1>
        <button className="primary-button" onClick={handleClick}>
          {authToken ? "Signup" : "Create Account"}
        </button>
        {showModal && (
          <AuthModal
            setShowModal={setShowModal}
            isSignUp={isSignUp}
            setIsSignUp={setIsSignUp}
          />
        )}
      </div>
    </div>
  );
};

export default Home;
