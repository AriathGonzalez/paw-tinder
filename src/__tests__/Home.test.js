import React from "react";
import { render, screen, act } from "@testing-library/react";
import "@testing-library/jest-dom";
import Home from "../pages/Home";

describe("React App", () => {
  test("Test renders text", () => {
    render(<Home />);
    const text = screen.getByText("Swipe RightÂ®");
    expect(text).toBeInTheDocument();
  });

  test("Renders Correctly", () => {
    const { container } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
