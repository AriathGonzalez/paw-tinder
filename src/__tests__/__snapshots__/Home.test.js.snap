// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React App Home Renders Correctly 1`] = `
<div
  class="overlay"
>
  <nav>
    <div
      class="logo-container"
    >
      <img
        alt="Logo"
        class="logo"
        src="paw_tinder_logo_black.png"
      />
    </div>
    <button
      class="nav-button"
    >
      Log In
    </button>
  </nav>
  <div
    class="home"
  >
    <h1
      class="primary-title"
    >
      Sniff RightÂ®
    </h1>
    <button
      class="primary-button"
    >
      Create Account
    </button>
  </div>
</div>
`;
